### PR TITLE
dcmtk: provide official CMake imported targets + bump deps + properly deprecate openjpeg/libsndfile options

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

File: recipes/dcmtk/all/conanfile.py [hunk 1]
```diff
@@ -63,8 +66,10 @@ def configure(self):
         if self.settings.os == "Windows":
             del self.options.with_tcpwrappers
         # Looking into source code, it appears that OpenJPEG and libsndfile are not used
```
Comment (prince-chrismc) [inline recipes/dcmtk/all/conanfile.py:68]: I can confirm the OpenJPEG is not being used

File: recipes/dcmtk/all/conanfile.py [hunk 2]
```diff
@@ -167,147 +170,116 @@ def package(self):
...
+            "ofstd"   : charset_conversion(),
+            "oflog"   : ["ofstd"],
+            "dcmdata" : ["ofstd", "oflog"] + zlib(),
+            "i2d"     : ["dcmdata"],
+            "dcmimgle": ["ofstd", "oflog", "dcmdata"],
+            "dcmimage": ["oflog", "dcmdata", "dcmimgle"] + png() + tiff(),
+            "dcmjpeg" : ["ofstd", "oflog", "dcmdata", "dcmimgle", "dcmimage", "ijg8", "ijg12", "ijg16"],
+            "ijg8"    : [],
+            "ijg12"   : [],
+            "ijg16"   : [],
```
Comment (prince-chrismc) [inline recipes/dcmtk/all/conanfile.py:234]: should the order be flipped?

File: recipes/dcmtk/all/conanfile.py [hunk 3]
```diff
@@ -167,147 +170,116 @@ def package(self):
...
+
+                if target_lib == "ofstd":
+                    if self.settings.os == "Windows":
+                        self.cpp_info.components[target_lib].system_libs.extend([
+                            "iphlpapi", "ws2_32", "netapi32", "wsock32"
+                        ])
+                    elif self.settings.os == "Linux":
+                        self.cpp_info.components[target_lib].system_libs.append("m")
+                        if self.options.with_multithreading:
+                            self.cpp_info.components[target_lib].system_libs.append("pthread")
```
Comment (prince-chrismc) [inline recipes/dcmtk/all/conanfile.py:280]: Could this be moved to the package_info scope? seems odd to cover over this for every lib

File: recipes/dcmtk/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/dcmtk/all/conanfile.py:234]: What do you mean?

File: recipes/dcmtk/all/conanfile.py (hunk 3 above)
Comment (SpaceIm) [reply recipes/dcmtk/all/conanfile.py]: moved outside the loop

File: recipes/dcmtk/all/conanfile.py (hunk 2 above)
Comment (prince-chrismc) [reply recipes/dcmtk/all/conanfile.py:234]: It seems odd to require components which have no been created, perhaps I am not following this correctly?

Comment (SpaceIm) [reply recipes/dcmtk/all/conanfile.py:234]: Order of creation of conan components doesn't matter. Moreover here it's a dictionnary, there is no order.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]
don't know if I like "deprecated" option values...

**Review by madebr — APPROVED** [review]

**Comment by madebr** [comment]
> 
> 
> don't know if I like "deprecated" option values...

`deprecated` is self-explanatory. Alternatively, one can use `None`.

**Pull request merged**